src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx

```diff
@@ -57,44 +86,64 @@ function QRCodeScannerScreen() {
   const handleBarCodeScanned = async (result: BarCodeScannerResult) => {
     const caseId = result.data;
     if (!validIds.includes(caseId)) {
-      // TODO: Display error toast message
-      setToast('Not a valid QRCODE!');
+      Toast.show({
+        type: 'error',
+        text1: 'Not a valid QRCODE!',
+        visibilityTime: 1000,
+      });
     } else if (userIds.includes(caseId)) {
-      setToast('DUPLICATES NOT ALLOWED!');
+      Toast.show({
+        type: 'error',
+        text1: 'Duplicate cases not allowed!',
+        visibilityTime: 1000,
+      });
     } else if (!scanned) {
       const caseData: Case = await getCaseById(caseId);
-      const { id, title, imageUrl, date, lawFirm, summary } = caseData;
-      router.push({
-        pathname: '/AllCases/QRCodeScanner/AddCase',
-        params: {
-          id,
-          title,
-          imageUrl,
-          date,
-          lawFirm,
-          summary,
-        },
-      });
+      setUserCase(caseData);
       setScanned(true);
-      setToast('');
     }
   };
 
+  const routeToAddCase = () => {
+    if (!userCase) {
+      return;
+    }
+    router.push({
+      pathname: `/AllCases/QRCodeScanner/AddCase/${userCase.id}`,
+      params: {
+        id: userCase.id,
+        title: userCase.title,
+        imageUrl: userCase.imageUrl,
+        date: userCase.date,
+        lawFirm: userCase.lawFirm,
+        summary: userCase.summary,
+      },
+    });
+  };
+
   if (hasPermission === permissions.DENIED) {
     return <Text>CANNOT ACCESS CAMERA!</Text>;
   }
 
   return (
     <View style={styles.container}>
-      <Text>Add a new case</Text>
+      <Text style={styles.topText}>Point your Camera at the QR code.</Text>
       <BarCodeScanner
         onBarCodeScanned={handleBarCodeScanned}
         style={[styles.scanner]}
       />
-      <Text style={styles.errorMessage}>{toast}</Text>
-      <TouchableOpacity onPress={() => router.back()} style={styles.button}>
-        <Text>Go Back</Text>
-      </TouchableOpacity>
+
+      {userCase && (
+        <TouchableOpacity
```

Comment:
The View Case button should be at the bottom of the screen rather than underneath the camera view (refer to Figma).